Feat/dv 67/create offer page

vadymchornyidev opened the pull request:
> Maryna made changes because we did not have time to finish the task to the demo

Commit:
add some styles

Commit:
add styles and media queries

Commit:
add private route

Commit:
add page with styles and media queries

Grasmus commented on src/assets/scss/assets/pages/_offerPage.scss:
```diff
@@ -0,0 +1,134 @@
+}
+
+@media only screen and (min-width: $max_tablet_size) and (max-width: $max_laptop_size) {
+
```
remove empty line

Commit:
remove empty line

Grasmus commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,95 @@
+export default function Offer() {
+    const { isLoaded } = useJsApiLoader({
+        id: "google-map-script",
+        googleMapsApiKey: "AIzaSyCU9Ghsq5fKRy9U8yWALIxnsBuxWIaLbhQ",
```
maybe, this key should be in constant

marynakernychna replied:
It can be taken out in constants

Grasmus commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,95 @@
+
+const center = {
+    lat: -3.745,
+    lng: -38.523,
```
it will be better if we avoid magic numbers

vadymchornyidev replied:
that's just for first image in maps ( THAT'S NEED )

marynakernychna replied:
> that's just for first image in maps ( THAT'S NEED )

So you have to add a comment to these magic numbers

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,95 @@
+import { GoogleMap, useJsApiLoader } from "@react-google-maps/api";
+import Header from "../navigation/header";
+
+const containerStyle = {
```
Why are styles written here? Can't they be taken out?

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,95 @@
+                            <select>
+                                Goods Category
+                                <option>Goods Category</option>
+                                <option>Їжа</option>
```
We now have a site in English

Commit:
add file for save api to google maps

Grasmus commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,95 @@
+                                <option>Одяг</option>
+                                <option>Гроші</option>
+                                <option>Запчастини</option>
+                                <option>Інше...</option>
```
maybe it should be in separated list variable, which will contain all goods categories

Commit:
resolve errors

Commit:
fix size

marynakernychna reviewed (commented):
> Recommendation not to edit files that are not relevant to your task in this case — src/api/configurations.js
> Recommendation not to make committees for changes on several lines, provided that these are not the last corrections
> Recommendation not to call the PR the full name of the branch

> No work with the server (logic of creating an offer)
> The problem with the map
> It is not possible to put a point to the map unless it is a public place, such as a restaurant, mall, etc.
> Why is Fortaleza, Brazil the default city on the map?


![image](https://user-images.githubusercontent.com/56841893/167460464-af70973c-8f56-4cfa-9ea7-360ba672c01e.png)

Commit:
git add new bug

Commit:
rewrite for antd

Commit:
stability version

marynakernychna committed:
Add styles, page elements & adjust request to the server

---Main---
> Add styles
> Add page elements
> Adjust request to the server

---Others---
> Make a page in a private route

Commit:
Merge branch 'feat/dv-67/create_offer_page' of https://github.com/IT-Academy-Social-Projects-KRV/Logistman-Client into feat/dv-67/create_offer_page

Commit:
work version

Commit:
add mobile version for offer page

Commit:
add geocoding

Commit:
change selection and add geocoding function

Commit:
add posibility to create offer from cart

Commit:
add google places autocomplete

Commit:
change GooglePlacesAutocomplete to custom AutoComplete

Commit:
add media queries and fix styles

Commit:
hide api key and clear code area

Commit:
fix datepicker and change styles for few components

Commit:
resolve conflicts

Commit:
change styles

Commit:
Stopped tracking env.local, and env.staging

Commit:
hide api key

YevheniiVovk commented on public/index.html:
```diff
@@ -10,7 +10,11 @@
+<script
+  src="https://maps.googleapis.com/maps/api/js?key=AIzaSyAdPX1hEy04ED1D8TLhLsTEwNSJ5xbo0Vo&libraries=places"></script>
+
+</html>
```
EOF

YevheniiVovk commented on src/assets/scss/assets/pages/_offerPage.scss:
```diff
@@ -0,0 +1,101 @@
+@media only screen and (max-width: $max_phone_size) {
+  .ant-form {
+    display: flex;
+    flex-direction: column !important;
```
Do NOT use !important

YevheniiVovk commented on src/assets/scss/assets/pages/_offerPage.scss:
```diff
@@ -0,0 +1,101 @@
+  }
+
+  .mapBlock {
+    height: 200px !important;
```
Do NOT use !important

Commit:
resolve issues

vadymchornyidev committed:
Merge branch 'dev' into feat/dv-67/create_offer_page

Commit:
give possibility to show error with rangepicker

vladtymo commented on public/index.html:
```diff
@@ -10,7 +10,13 @@
+  <div id="root">
+  </div>
+
+  <script
+    src="https://maps.googleapis.com/maps/api/js?key=AIzaSyAdPX1hEy04ED1D8TLhLsTEwNSJ5xbo0Vo&libraries=places">
+  </script>
```
I think it's better to move this code to a separate file/component. Let's investigate it...

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,347 @@
+        });
+      },
+      (error) => {
+        console.error(error);
```
Do not output the message in the console, tell the user that something went wrong

Grasmus replied:
We need this here to tell user that it should be unless 12 hours between start and expiration time

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,347 @@
+                },
+              ]}
+            >
+              <Input name="region" placeholder="Enter your settlement" />
```
Why do we have two fields for settlement?

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,347 @@
+                },
+              ]}
+            >
+              <Select placeholder="Select good categorie">
```
The word is misspelled

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,347 @@
+  fullscreenControl: true,
+};
+
+export default function Offer() {
```
It is better to call OfferPage, it is more in line with the name of the page

marynakernychna commented on src/constants/userRoles.js:
```diff
@@ -1,4 +1,6 @@
 export const userRoles = {
-    GUEST: 'Guest',
-    USER: 'User'    // when adding a role, add it to authReducer in the set access case
```
Return the comment

marynakernychna commented on src/reduxActions/auth/types.js:
```diff
@@ -1,2 +1,5 @@
 export const LOGOUT = "LOGOUT";
+export const SET_USER_ROLE = "SET_USER_ROLE";
+
```
Why is the empty blank?
You put commas where you don't need them, and you don't separate the code with empty lines, but you decided to put it here

marynakernychna commented on src/services/goodCategoryService.js:
```diff
@@ -0,0 +1,31 @@
+import { goodCategoryMessages } from "../constants/messages/goodCategory";
+import goodCategoryService from "../api/goodCategory";
+
+export async function getGoodCategories() {
```
Why do we need this function?

vadymchornyidev replied:
for get good categories in offer page.

marynakernychna replied:
We do this in the next function

marynakernychna commented on src/services/offerService.js:
```diff
@@ -0,0 +1,52 @@
+    startDate: values.startDate[0]._d,
+    expirationDate: values.startDate[1]._d,
+    goodCategory: values.goodCategory,
+    role: "SENDER",
```
You have constants for this

marynakernychna commented on src/services/offerService.js:
```diff
@@ -0,0 +1,52 @@
+      settlement: values.settlement,
+      region: values.region,
+      order: 0,
+      tripId: null,
```
You do not have to write this field at all

marynakernychna commented on src/services/offerService.js:
```diff
@@ -0,0 +1,52 @@
+      address: values.address,
+      settlement: values.settlement,
+      region: values.region,
+      order: 0,
```
You can create a constant under this, called "Order by default"

marynakernychna commented on src/services/offerService.js:
```diff
@@ -0,0 +1,52 @@
+            generalErrorMessages.SOMETHING_WENT_WRONG
+          );
+          if (err.response.data.errors[""][0]) {
+            errorMessage(
```
Why do we need to display two different error messages?

vadymchornyidev replied:
Because, I wanna show for user unaccepted errors

marynakernychna replied:
Reread the question

marynakernychna commented on src/services/pointService.js:
```diff
@@ -0,0 +1,38 @@
+import { pointErrorMessages } from "../constants/messages/point";
+import pointService from "../api/point";
+
+export function createPoint(values, history) {
```
Where do we use it? I did not find such a line

marynakernychna commented on src/services/pointService.js:
```diff
@@ -0,0 +1,38 @@
+import pointService from "../api/point";
+
+export function createPoint(values, history) {
+    var model = {
```
You can simply send values without creating a model, if they have the same fields

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,347 @@
+          <div className="leftFormBlock">
+            <Form.Item
+              name="address"
+              rules={[
```
I can enter anything in this line, are you sure this is a good validation and that it matches the one in the service?

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,347 @@
+            </Form.Item>
+            <Form.Item
+              name="settlement"
+              rules={[
```
I can enter anything in this line, are you sure this is a good validation and that it matches the one in the service?

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,347 @@
+            </Form.Item>
+            <Form.Item
+              name="region"
+              rules={[
```
I can enter anything in this line, are you sure this is a good validation and that it matches the one in the service?

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,347 @@
+              name="goodCategory"
+              rules={[
+                {
+                  type: "string",
```
You do not need this rule because you have Select

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,347 @@
+              <Select placeholder="Select good categorie">
+                {data?.map((res, idx) => (
+                  <Option value={res.name} key={idx}>
+                    {res.name}
```
I think it should be in lowercase

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,347 @@
+              ]}
+            >
+              <RangePicker
+                disabledDate={disabledDate}
```
Validation does not match the one in the service

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,347 @@
+
+            <Form.Item
+              name="goodsWeight"
+              rules={[
```
I can enter any number here, are you sure this is a good validation?

marynakernychna reviewed (commented):
No normal validation

![image](https://user-images.githubusercontent.com/56841893/169980360-6a171611-2984-414b-8952-d2a016e6ba99.png)

![image](https://user-images.githubusercontent.com/56841893/169980291-a7d633e8-192c-4702-b799-7ae50c763ac5.png)

PlacesAutocomplete works crookedly.

![image](https://user-images.githubusercontent.com/56841893/169983585-c510ad70-c96f-4ef1-a22f-b19228527053.png)

![image](https://user-images.githubusercontent.com/56841893/169983634-c14a4f58-181e-4fbd-be29-2d837b3d7573.png)

Commit:
remove no-need items

Grasmus commented on src/constants/offerValues.js:
```diff
@@ -0,0 +1,3 @@
+export const offerValues = {
+    ORDER_BY_DEFAULT: 0
+}
```
Add an empty line

Commit:
resolve errors

Commit:
remove no-need items

vadymchornyidev committed:
Merge branch 'dev' into feat/dv-67/create_offer_page

Commit:
hide api key

marynakernychna commented on src/App.js:
```diff
@@ -13,58 +13,50 @@ import PrivateRoute from "./privateRoute";
+  return (
+    <Router history={history}>
+      <Switch>
+        <PrivateRoute exact path="/main" allowedRoles={[userRoles.USER]}>
```
Revert to previous formatting

marynakernychna commented on src/App.js:
```diff
@@ -13,58 +13,50 @@ import PrivateRoute from "./privateRoute";
+          <UserProfilePage />
+        </PrivateRoute>
+
+        <PrivateRoute exact path="/offer" allowedRoles={[userRoles.USER]}>
```
Rename to "create-offer"

Grasmus commented on src/components/authentication/login/index.jsx:
```diff
@@ -84,14 +83,12 @@ function Login() {
-                            Remember me
-                        </Checkbox>
-                        <Link to="/forgotPassword">Forgot password</Link>
+                       
```
Drop this empty div

Grasmus commented on src/assets/scss/assets/variables/_colors.scss:
```diff
@@ -2,6 +2,7 @@ $green_color: #20DF7F;
 $body_color: #093545;
 $input_color: #224957;
 $header_color: #05192c;
+$light_gray: #A1A1A1;
```
This color already exists as a constant
![image](https://user-images.githubusercontent.com/91031013/170431249-9ffbd4d0-077e-4146-a04e-5c2f6948e04b.png)

Grasmus commented on src/components/navigation/header.jsx:
```diff
@@ -43,15 +43,7 @@ export default function Header() {
-                        <img src={theme} alt="theme-icon"/>
-                    </button>
-                </div>
+                <div className="support_block"></div>
```
Drop this empty div

Grasmus commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,356 @@
+};
+
+const center = {
+  lat: 50.64,
```
Remove this magic numbers

marynakernychna commented on src/constants/map.js:
```diff
@@ -0,0 +1,3 @@
+export const googleApi = {
+    KEY: "AIzaSyAdPX1hEy04ED1D8TLhLsTEwNSJ5xbo0Vo"
+}
```
Empty line at EoF

marynakernychna commented on src/constants/map.js:
```diff
@@ -0,0 +1,3 @@
+export const googleApi = {
+    KEY: "AIzaSyAdPX1hEy04ED1D8TLhLsTEwNSJ5xbo0Vo"
```
The key is public

Grasmus commented on src/constants/messages/offer.js:
```diff
@@ -0,0 +1,7 @@
+    CREATE_OFFER_SUCCESS: "Offer successfully created!",
+    CREATE_OFFER_FAILED: "Create offer failed!",
+    ENTER_ALL_INPUTS: "You got this error because you did not fill in all the fields",
+    MAP_IS_NOT_WORK: "Map is not work, please wait!",
```
Incorrect. Map does not work

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -117,8 +118,11 @@ export default function Offer() {
+      (error)=>{
+        errorMessage(
+          offerErrorMessages.CREATE_OFFER_FAILED,
+          offerErrorMessages.MAP_IS_NOT_WORK
```
The message: "Map is not work, please wait!". But the map will not load, because there will be an error

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -117,8 +118,11 @@ export default function Offer() {
       },
-      (error) => {
-        console.error(error);
+      (error)=>{
```
This field is not used

Grasmus commented on src/constants/messages/offer.js:
```diff
@@ -0,0 +1,7 @@
+    CREATE_OFFER_FAILED: "Create offer failed!",
+    ENTER_ALL_INPUTS: "You got this error because you did not fill in all the fields",
+    MAP_IS_NOT_WORK: "Map is not work, please wait!",
+    TIME_INTERVAL_INCORRECT: "Time interval must be minimum 12 hourst between start date and expiration date!"
```
*hours

Grasmus commented on src/constants/messages/offer.js:
```diff
@@ -0,0 +1,7 @@
+export const offerErrorMessages = Object.freeze({
+    CREATE_OFFER_SUCCESS: "Offer successfully created!",
+    CREATE_OFFER_FAILED: "Create offer failed!",
```
Incorrect. Failed to create an offer

marynakernychna commented on src/constants/messages/offer.js:
```diff
@@ -1,5 +1,7 @@
-    ENTER_ALL_INPUTS: "You got this error because you did not fill in all the fields"
+    ENTER_ALL_INPUTS: "You got this error because you did not fill in all the fields",
+    MAP_IS_NOT_WORK: "Map is not work, please wait!",
+    TIME_INTERVAL_INCORRECT: "Time interval must be minimum 12 hourst between start date and expiration date!"
```
Rename the constant

Grasmus commented on src/constants/messages/offer.js:
```diff
@@ -0,0 +1,7 @@
+export const offerErrorMessages = Object.freeze({
+    CREATE_OFFER_SUCCESS: "Offer successfully created!",
+    CREATE_OFFER_FAILED: "Create offer failed!",
+    ENTER_ALL_INPUTS: "You got this error because you did not fill in all the fields",
```
It is better to output: "All fields must be filled"

Grasmus commented on src/constants/messages/point.js:
```diff
@@ -0,0 +1,4 @@
+export const pointErrorMessages = Object.freeze({
+    CREATE_POINT_FAILED: "Create point failed!",
```
Incorret. Failed to create a point

marynakernychna commented on src/services/offerService.js:
```diff
@@ -1,23 +1,25 @@
 import { errorMessage, successMessage } from "./alert.service";
```
We already have a service for offers, move everything there

Grasmus commented on src/constants/userRoles.js:
```diff
@@ -1,4 +1,6 @@
-}
+    GUEST: "Guest",
+    USER: "User", // when adding a role, add it to authReducer in the set access case
+    RECIPIENT: "Recipient",
```
Is it better to separate this two roles (Recipient and Sender) in offerRoles constant?

marynakernychna commented on src/api/point.js:
```diff
@@ -0,0 +1,8 @@
+import { POINT_URLS } from "../constants/url";
```
We do not use this service anywhere

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,356 @@
+                },
+              ]}
+            >
+              <Input type="number" min="0" addonAfter="kg" placeholder="Goods Weight" />
```
Show the user about the error immediately, not after he presses the button

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,356 @@
+              name="goodsWeight"
+              rules={[
+                {
+                  message: inputValidationErrors.EMPTY_GOOD_WEIGHT_MESSAGE,
```
For what?

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,356 @@
+  fullscreenControl: true,
+};
+
+export default function OfferPage() {
```
After loading the page 3 error messages at once

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,356 @@
+              className="description"
+              rules={[
+                {
+                  message: inputValidationErrors.EMPTY_DESCRIPTION_MESSAGE,
```
For what?

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,356 @@
+                },
+              ]}
+            >
+              <PlacesAutocomplete
```
When choosing a location, the error is that I can't create an offer

marynakernychna replied:
![image](https://user-images.githubusercontent.com/56841893/170441841-08b9b802-5cff-4414-900f-c6989e65e395.png)

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,356 @@
+            >
+              <Input type="number" min="0" addonAfter="kg" placeholder="Goods Weight" />
+            </Form.Item>
+            <Form.Item
```
Make description field height dependent on the inner text

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,356 @@
+              </Select>
+            </Form.Item>
+
+            <Form.Item
```
![Screenshot 2022-05-26 104334](https://user-images.githubusercontent.com/56841893/170442387-768b571c-9cad-483f-9b06-e0057cc33b49.jpg)

I can specify the time in the past

marynakernychna commented on src/components/offerPage/index.jsx:
```diff
@@ -0,0 +1,356 @@
+
+          <div className="rightFormBlock">
+            <div className="mapBlock">
+              <GoogleMap
```
Maps do not work, the point is put, but the coordinates are not set

marynakernychna commented on src/components/mainPage/index.jsx:
```diff
@@ -1,15 +1,17 @@
                 <div className="role-block">
                     <div className="role block-sender">
-                        <span>I can donate some goods</span>
+                        <Link className="role-link" to="/offer">
```
Why go to the page only by clicking on the text and not on the whole tile?

vladtymo commented on src/constants/messages/goodCategory.js:
```diff
@@ -0,0 +1,3 @@
+export const goodCategoryMessages = Object.freeze({
+    GET_GOOD_CATEGORIES__FAILED: "Get Good Categories failed!"
```
Correct the sentence grammatically.

Commit:
change styles for form

Commit:
add adaptive for tablet

Commit:
merge two files with offers
